src/systems/inter_mesh_projection.C

```diff
@@ -18,14 +18,25 @@
 // Local includes
 #include "libmesh/inter_mesh_projection.h"
 
+#include "libmesh/system_norm.h"
+#include "libmesh/enum_norm_type.h"
+
+
 // C++ includes
 
 namespace libMesh
 {
+    GradientFunction::GradientFunction(MeshFunction * _mesh_function)
+    {
+     //mesh_function = std::unique_ptr<MeshFunction>(dynamic_cast<MeshFunction *>((_mesh_function->clone()).get()));
```

Comment:
You were on a better track with this one.  We know we will always have a MeshFunction in mesh_function, so let's do that, then we can do the casting just once, when the unique_ptr is assigned, not every single time it's used.  Let's also use libmesh_cast_ptr, so it's not doing a dynamic_cast in opt mode or on systems where RTTI is disabled.

Reply:
Yes, that does make sense.

Reply:
Fixed in an upcoming commit.